core/src/avm2/object.rs

```diff
@@ -207,19 +207,75 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
         receiver.init_property_local(receiver, name, value, activation)
     }
 
+    /// Determines if a local slot already exists and is occupied.
+    fn has_slot_local(self, id: u32) -> bool;
+
     /// Retrieve a slot by its index.
-    fn get_slot(self, id: u32) -> Result<Value<'gc>, Error>;
+    #[allow(unused_mut)]
+    fn get_slot(
+        mut self,
+        activation: &mut Activation<'_, 'gc, '_>,
+        id: u32,
+    ) -> Result<Value<'gc>, Error> {
+        if !self.has_slot_local(id) {
+            let slot_trait = self
+                .get_trait_slot(id)?
+                .ok_or_else(|| format!("Slot index {} out of bounds!", id))?;
+            self.install_trait(activation, slot_trait, self.into())?;
+        }
+
+        self.get_slot_local(id)
+    }
+
+    /// Retrieve a slot by its index, ignoring uninstalled traits.
+    fn get_slot_local(self, id: u32) -> Result<Value<'gc>, Error>;
 
     /// Set a slot by its index.
+    #[allow(unused_mut)]
     fn set_slot(
+        mut self,
```

Comment:
These are not unused muts. We need to call `install_trait` which is a mutable borrow method inside of the default implementation of this method. For whatever reason Clippy doesn't recognize this so I just `#[allow]`ed it.

The other option would be to force these to be borrowed mutable rather than owned mutable if you wanted to avoid the `#[allow]`; or change all of the called `&mut self` methods to instead take `self` so that I don't get an error when trying to call them.